app/src/main/java/com/chess/clock/fragments/TimeControlFragment.java

```diff
@@ -150,38 +179,61 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
+        mSameAsPlayerOneSwtich = (SwitchCompat) v.findViewById(R.id.switch_same_as_player_one);
+        mSameAsPlayerOneSwitchContainer.setVisibility(GONE);
+        ((SettingsActivity) getActivity()).setBottomNavigationViewVisibility(VISIBLE);
+        ((SettingsActivity) getActivity()).setNavigationOnItemSelectedListener(
+            new OnNavigationItemSelectedListener() {
+                @Override
+                public boolean onNavigationItemSelected(
+                    @NonNull MenuItem item) {
+                    switch (item.getItemId()) {
+                        case R.id.nav_player1:
+                            mSameAsPlayerOneSwitchContainer.setVisibility(GONE);
+                            break;
+                        case R.id.nav_player2:
+                            mSameAsPlayerOneSwitchContainer.setVisibility(VISIBLE);
+                            break;
+                    }
+                    mplayerOneSelected = !mplayerOneSelected;
+                    mSelectedTimeControl = mplayerOneSelected ? mTimeControlWrapper.getTimeControlPlayerOne() : mTimeControlWrapper.getTimeControlPlayerTwo();
+                    StageAdapter stageAdapter = new StageAdapter(getActivity(), mSelectedTimeControl.getStageManager(), TimeControlFragment.this);
+                    mStageListView.setAdapter(stageAdapter);
+                    mTimeIncrementDescription.setText(mSelectedTimeControl.getTimeIncrement().toString());
+                    updateDisplay();
+                    return true;
+                }
+            });
 
         if (mStageListView != null) {
 
             mStageListView.setOnItemClickListener(mItemClickListener);
-            if (mTimeControl != null) {
+            if (mTimeControlWrapper != null) {
 
                 if (savedInstanceState != null) {
                     mTimeControlSnapshot = savedInstanceState.getParcelable(STATE_TIME_CONTROL_SNAPSHOT_KEY);
                 } else {
                     // Save copy to check modifications before exit.
-                    try {
-                        mTimeControlSnapshot = (TimeControl) mTimeControl.clone();
-                    } catch (CloneNotSupportedException e) {
-                        e.printStackTrace();
-                    }
+                    mTimeControlSnapshot = new TimeControlWrapper(mTimeControlWrapper.getTimeControlPlayerOne(), mTimeControlWrapper.getTimeControlPlayerTwo());
                 }
 
                 // Setup Time Control Name Edit Text
                 mTimeControlNameEditText = (EditText) v.findViewById(R.id.time_control_name);
                 mTimeControlNameEditText.addTextChangedListener(mTextWatcher);
 
-                if (mTimeControl.getName() != null && !mTimeControl.getName().equals("")) {
-                    mTimeControlNameEditText.setText(mTimeControl.getName());
+                TimeControl tc = mTimeControlWrapper.getTimeControlPlayerOne();
```

Comment:
Is getting player one's time controls always going to be correct?

Reply:
I figured that before when we were using one time control we used it as the source of truth. So with the addition of the time control wrapper, the player one time control will be the source of truth when the fragment is first created.